Fix the link on the upstream chart legend

rdwebdesign opened the pull request:
### What does this PR aim to accomplish?

When 2 or more upstream servers have the same name, both links on the chart legend point to the same IP:

<img width="600" alt="Upstreams" src="https://github.com/user-attachments/assets/31352cb3-6ed3-4341-b07a-58c50b495f20" />

Each link should point to the correct IP and show different queries, but the issue happens because the current code uses an object containing the server name (label) as `key` and the `value` is the IP, but when there are 2 items with the same key, the first one is overwritten and its IP is lost.

### How does this PR accomplish the above?

Using an array of IPs with indexes will avoid the issue.

This PR also adds the IP to the legend tooltip, to make easier for users to identify different DNS servers, even when they have the same name:

<img width="500" alt="image" src="https://github.com/user-attachments/assets/82cb508c-c6fe-438d-aced-ade1a454a4f9" />


### Link documentation PRs if any are needed to support this PR:

none

---
**By submitting this pull request, I confirm the following:**

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against.
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
6. I have checked that another pull request for this purpose does not exist.
7. I have considered, and confirmed that this submission will be valuable to others.
8. I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
9. I give this submission freely, and claim no ownership to its content.

---
- [x] I have read the above and my PR is ready for review. *Check this box to confirm*

rdwebdesign committed:
Remove unnecessary code

The outer `if` (line 92) already guarantees only 2 possible values.
If `isQueryTypeChart` is false, `isForwardDestinationChart` must be true.

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Use an array to store upstream server IPs

This will avoid overwritting the IP when more than one upstream DNS server uses the same name

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Show the upstream server IP on the title tooltip

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

yubiuser commented:
This is really good. 

Could you also try to fix #3 I've found [here](https://github.com/pi-hole/FTL/issues/2443#issuecomment-2993369067)

> If one comes from the dashboard, the the filter does not pre-select the existing entry in the drop-down but creates a new one.

![Image](https://github.com/user-attachments/assets/503fff69-ca76-4a73-bf73-3a56038b1a31)

rdwebdesign commented:
Hmmm, I hadn't noticed that item 3 in your previous comment...

I will try to fix this tonight.

rdwebdesign committed:
Include the upstream DNS server name to the link, when needed

The link must send the "upstream" parameter in exactly the same format used
by the "suggestions" API.
The format is: "upstream=<IP>#<port> (<servername>)".
This will ensure that when a link is clicked, the correct server name will
be highlighted in the SELECT element on the queries.lp page and no other
OPTION element will be created.

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign commented:
I already changed the code.

yubiuser reviewed (commented):
That's working really well now.
Last nitpick: The tooltip of the doughnut chart contains `name#port: percentage` but does not contain IP. So for the same upstream name one can only distinguish by color. Could you include the IP here as well?

rdwebdesign commented:
I'm not sure if we need to include the IP to the graphic tooltip.

<img width="333" alt="image" src="https://github.com/user-attachments/assets/23d8c821-3fcf-469e-b6af-8c841074dd99" />  
  
<img width="335" alt="image" src="https://github.com/user-attachments/assets/96113028-048a-41a8-a313-2531012e2391" />

I think the tooltip will be too wide if the server name and IP are both long. 

And adding a line break doesn't look good:
<img width="316" alt="image" src="https://github.com/user-attachments/assets/d3f39396-cb79-4b9e-af32-d76a5a27b1b4" />

yubiuser commented:
Yeah, I feared this. But without including the IP the only way to distinguish the same names is by color - which affects accessibility :-/

rdwebdesign commented:
Maybe we should add a more distinguishable effect, like this:

![hover_effect](https://github.com/user-attachments/assets/fce35c80-3b81-4eba-800c-24a3e8f4942b)

yubiuser commented:
Nice!

rdwebdesign committed:
Add offset effect on hover to the doughnut charts

We need to add a small padding to avoid "clipping" the arc/slice.
This happens because when an arc/slice expands, it grows beyond the canvas
limits and get clipped.

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Trigger the offset effect when the mouse is over a legend item

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

yubiuser reviewed (commented):
I observed an interesting effect: the sensitive 'mouseover' area is as big as the longest element in the the whole list. For shorter elements, this means the effect will also apply even when the mouse is not above the text.

![Image](https://github.com/user-attachments/assets/90666c73-f4ba-4407-aac3-51be02cab408)

rdwebdesign commented:
All elements of each legend have the same width:
<img width="1247" height="345" alt="image" src="https://github.com/user-attachments/assets/032ee445-60e2-408e-a638-a5bd9ebfdfb4" />

This is exactly what we have in master and development branches (you can try to click on the right on shorter item to test)

I really don't think this is an issue and this is just cosmetics.

If we decide to change it, we can open a new PR to change the CSS style, to show a background color, border, whatever... 

Maybe something like this:
<img width="200" alt="image" src="https://github.com/user-attachments/assets/d42678fe-d4d5-4df3-8b78-b31c0f58c1f7" />

yubiuser reviewed (approved):
I think it's only a small detail, no need to fix it now.
